query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/ResultSetTypePredicateProvider.java

```diff
@@ -0,0 +1,21 @@
+package org.hypertrace.core.query.service.postgres;
+
+import java.sql.ResultSet;
+
+/**
+ * This interface is used to determine which handler will parse the Postgres ResultSet in
+ * PostgresBasedRequestHandler#convert(). We define it to make it easy to unit test the parsing
+ * logic since: - The implementations of ResultSet are package private and there's no way to
+ * determine the concrete type of the ResultSet object other than using the class name. See
+ * DefaultResultSetTypePredicateProvider class. - The ResultSet interface itself is implemented non
+ * uniformly by its implementations. The defined methods in the interface do not return consistent
+ * data across the implementations and the format of the implementations is different. - However,
+ * since it seems like for "sql" format the ResultTableResultSet is being returned for all Postgres
+ * query types we might be able to get rid of this in the future and have a single flow to parse the
+ * Postgres Response.
+ */
+public interface ResultSetTypePredicateProvider {
```

Comment:
I read above comment, but guess, need to check with you offline to understand this one.